src/@types/Modal.d.ts

```diff
@@ -0,0 +1,28 @@
+export interface ModalProps extends OnModalProps {
+  children: ReactNode;
+}
+
+export interface OnModalProps {
+  onModal: onModal;
+}
+export interface ModalShareProps {
+  userId: number;
+  folderId: number;
+  folderName: string;
+}
+
+export interface ShareButtonProps {
+  src: string;
+  text: string;
+  alt: string;
+  shareUrl: string;
+}
+
+export interface ShareImgDivProps {
+  src: string;
+  alt: string;
+  text: string;
+  handleClick?: () => void;
+}
+
+export type onModal = (type?: string, link?: string) => void;
```

Comment:
불필요한 옵셔널은 좋지 않습니다!

그리고 type은 없고 link는 넣어야 되는 경우에는 어떻게 적용해야 할까요?

이런 경우에는 아래와 같이 객체 형태로 묶으시는 편이 좋습니다.
또한 모두 옵셔널인 경우 Partial 을 사용해보세요!
```suggestion
export type onModal = (params: Partial<{ type: string; link: string }>) => void;
```